x/programs/rust/wasmlanche_sdk/src/memory.rs

```diff
@@ -68,6 +71,73 @@ impl Memory {
     }
 }
 
+/// `SmartPtr` is an i64 where the first 4 bytes represent the length of the bytes
+/// and the following 4 bytes represent a pointer to WASM memeory where the bytes are stored.
+pub type SmartPtr = i64;
```

Comment:
Do you want to create an issue to rename this? Smart-pointers are already thing in Rust [^1]. An `Arc` is an example of a smart-pointer. While there are tons of terms that are overloaded, I wouldn't want a new program dev to land in this code for whatever reason and get confused. 

[^1]: https://doc.rust-lang.org/book/ch15-00-smart-pointers.html

Reply:
yea, overloading`SmartPtr` isn't the best but every other name felt not as good. Maybe `argPtr` or `memoryPtr`? will create an issue.

Reply:
What about... wait for it... `Ptr` or `HostPtr`? I also like `ArgPtr` and `MemPtr`. The only reason I don't like `SmartPtr` is because it immediately makes me think that the type is something it isn't, whereas all the other names are equally nebulous and I therefore have to look at the implementation. 

🤷 , I saw you created the issue. We can deal with this later